### ISSUE TITLE
Size is unajustable.

### ISSUE THREAD

**Issue by rainwashed** [dmjones/flutter_auth_buttons#31]
Tried everything, flex, SizedBox, etc. It just keeps staying at that constant width and I can't change it or adjust it to what I need it to be.

**Comment by sam-parks** [comment]
I am also running into this issue. 

Apple requires 3rd party login options to be the same size so I can't release my app to the App Store.

**Comment by dmjones** [comment]
Can you provide a small, self-contained example that demonstrates your problem?

**Comment by sam-parks** [comment]
I am using the GoogleSignInButton and the AppleSignInButton: 

<img width="352" alt="Screen Shot 2020-07-10 at 9 38 36 AM" src="https://user-images.githubusercontent.com/44235716/87162265-aefcb700-c293-11ea-9b2f-c6f6cc9e1d2c.png">


```
Container(
            padding: SizeConfig.screenHeight <= 640
                ? EdgeInsets.zero
                : EdgeInsets.only(top: 20),
            child: GoogleSignInButton(
              onPressed: () {
                _signInBloc.add(SignInGooglePressed());
              },
              darkMode: true,
            ),
          ),
          if (_appleSignIn)
            Padding(
              padding: SizeConfig.screenHeight <= 640
                  ? EdgeInsets.zero
                  : EdgeInsets.only(top: 20),
              child: AppleSignInButton(
                style: AppleButtonStyle.whiteOutline,
                onPressed: () {
                  _signInBloc.add(SignInApplePressed());
                },
              ),
            ),


```

**Comment by dmjones** [comment]
I'm reopening this issue, since it is still active. @CrystalxCode you should be able to unsubscribe from notifications if you don't want to hear more about it.

**Comment by vrijmetse** [comment]
I tried to use `IntrinsicWidth` but buttons dissapeared.

```
IntrinsicWidth(
                  child: Column(
                    mainAxisSize: MainAxisSize.min,
                    crossAxisAlignment: CrossAxisAlignment.stretch,
                    children: <Widget>[
                      FacebookSignInButton(
                          text: "Connect with Facebook",
                          onPressed: () {
                            Navigator.push(
                              context,
                              MaterialPageRoute(
                                builder: (context) => SocialLoginScreen(
                                  provider: "Facebook",
                                ),
                              ),
                            );
                          }),
                      GoogleSignInButton(
                        text: "Connect with Google",
                        onPressed: () {
                          Navigator.push(
                            context,
                            MaterialPageRoute(
                              builder: (context) => SocialLoginScreen(
                                provider: "Google",
                              ),
                            ),
                          );
                        },
                      ),
                    ],
                  ),
                );
```

**Comment by MrLightful** [comment]
@dmjones can somebody please tackle this issue? It looks pretty bad with all buttons having different sizes.

Edit: nvm. 👇

**Comment by dmjones** [comment]
Can you reproduce your problem using a `RaisedButton`?

One goal of this package is to create buttons that can be styled like standard Flutter buttons. If you can demonstrate that buttons from this package behave differently to a `RaisedButton`, that would be a bug.

To get buttons with the same width you can place them in a column with `crossAxisAlignment: CrossAxisAlignment.stretch`. Here's a modified version of the [example code](https://github.com/dmjones/flutter_auth_buttons/blob/master/example/lib/main.dart) that does that:

```flutter
import 'package:flutter/material.dart';
import 'package:flutter_auth_buttons/flutter_auth_buttons.dart';

void main() async {
  runApp(new MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const padding = 25.0;

    return MaterialApp(
      title: 'Button Demo',
      home: Scaffold(
        appBar: AppBar(
          title: Text("flutter_auth_buttons"),
        ),
        backgroundColor: Color.fromARGB(0xFF, 0xF0, 0xF0, 0xF0),
        body: SingleChildScrollView(
          child: Padding(
            padding: EdgeInsets.only(left: padding, right: padding),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: <Widget>[
                SizedBox(height: padding),
                AppleSignInButton(onPressed: () {}),
                AppleSignInButton(
                    onPressed: () {}, style: AppleButtonStyle.whiteOutline),
                AppleSignInButton(
                    onPressed: () {}, style: AppleButtonStyle.black),
                SizedBox(height: padding),
                GoogleSignInButton(onPressed: () {}),
                GoogleSignInButton(onPressed: () {}, darkMode: true),
                SizedBox(height: padding),
                FacebookSignInButton(onPressed: () {}),
                SizedBox(height: padding),
                TwitterSignInButton(onPressed: () {}),
                SizedBox(height: padding),
                MicrosoftSignInButton(onPressed: () {}),
                MicrosoftSignInButton(onPressed: () {}, darkMode: true),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

Result:

<img width="348" alt="image" src="https://user-images.githubusercontent.com/3658671/89072205-0ad8ce00-d370-11ea-853b-cdb8a23c18da.png">

**Comment by frederickhazel6996** [comment]
try placing the button in a container and give the container a width

` Container(
                    width: 300,
                    child: GoogleSignInButton(
                      onPressed: () => {},
                      darkMode: false,
                    ),
                  )`

**Comment by Jwildsmith** [comment]
@dmjones The solution you presented above still does not fully solve the issue. The insides of the buttons are off centered which goes against a lot of design guidelines.

The needed solution would be for the Icon and text within the button to be perfectly centered inside the button as the width grows. You can keep the correctly formatted padding between the text and icon and it will look right as long as the whole content of the button remains centered.

See how these buttons are wierdly scaled, particularly the Facebook one:
![image](https://user-images.githubusercontent.com/22238134/90213652-f1963f80-ddbb-11ea-9677-1d6a07b84173.png)

**Comment by dmjones** [comment]
@Jwildsmith Good idea. A backwards-compatible solution would be to add a boolean "centered" property that defaults to false.

PRs welcome, as I don't have much time to work on this right now.

**Comment by Jwildsmith** [comment]
@dmjones Opened a PR that should fix this issue and needs review

**Comment by dmjones** [comment]
This is now merged in and released as 0.10.0. I'll close this issue for now, but feel free to reopen if needed.
